src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,222 @@
+  rmul_comm_rmul_rmul := λ a b, by rw [← mul_assoc, ← mul_assoc], }
+
+@[priority 100] -- see Note [lower instance priority]
+instance comm_semigroup.is_comm_jordan [comm_semigroup A] : is_comm_jordan A :=
+{ mul_comm := mul_comm,
+  jordan := λ a b, mul_assoc _ _ _, }
+
+local notation `L` := add_monoid.End.mul_left
+local notation `R` := add_monoid.End.mul_right
+
+/-! The Jordan axioms can be expressed in terms of commuting multiplication operators -/
+section lie
+variables {A} [non_unital_non_assoc_ring A] [is_jordan A]
+
+@[simp] lemma lie_lmul_rmul (a : A) : ⁅L a, R a⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_comm_rmul _ _).symm
+
+@[simp] lemma lie_lmul_lmul_sq (a : A) : ⁅L a, L (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_lmul_comm_lmul _ _).symm
+
+@[simp] lemma lie_lmul_rmul_sq (a : A) : ⁅L a, R (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_comm_rmul_rmul _ _).symm
+
+@[simp] lemma lie_lmul_sq_rmul (a : A) : ⁅L (a * a), R a⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_lmul_comm_rmul _ _)
+
+@[simp] lemma lie_rmul_rmul_sq (a : A) : ⁅R a, R (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.rmul_comm_rmul_rmul _ _).symm
+
+end lie
+
+variables [non_unital_non_assoc_ring A] [is_comm_jordan A]
+
+/-- Linearise the operator form of the Jordan axiom (`lie_L_L_sq`) by substituting in a → a + b and
+expanding. -/
+lemma lie_L_L_sq_add_lie_L_L_sq_add_two_lie_L_L_mul_add_two_lie_L_L_mul (a b : A) :
+  ⁅L a, L (b * b)⁆ + ⁅L b, L (a * a)⁆ + (2:ℤ)•⁅L a, L (a * b)⁆ + (2:ℤ)•⁅L b, L (a * b)⁆ = 0 :=
+begin
+  symmetry,
+  calc 0 = ⁅L (a + b), L ((a + b) * (a + b))⁆ : by rw (lie_lmul_lmul_sq (a + b))
+    ... = ⁅L a + L b, L (a * a + a * b + (b * a + b * b))⁆ :
+      by rw [add_mul, mul_add, mul_add, map_add]
+    ... = ⁅L a + L b, L (a * a) + L(a * b) + (L(a * b) + L(b * b))⁆ :
+      by rw [map_add, map_add, map_add, is_comm_jordan.mul_comm b a]
+    ... = ⁅L a + L b, L (a * a) + (2:ℤ)•L(a * b) + L(b * b)⁆ :
+      by rw [ two_smul, ← add_assoc, ← add_assoc]
+    ... = ⁅L a, L (a * a)⁆ + ⁅L a, (2:ℤ)•L(a * b)⁆ + ⁅L a, L(b * b)⁆
+      + (⁅L b, L (a * a)⁆ + ⁅L b,(2:ℤ)•L(a * b)⁆ + ⁅L b,L(b * b)⁆) :
+        by rw [add_lie, lie_add, lie_add, lie_add, lie_add]
+    ... = (2:ℤ)•⁅L a, L (a * b)⁆ + ⁅L a, L(b * b)⁆ + (⁅L b, L (a * a)⁆ + (2:ℤ)•⁅L b,L(a * b)⁆) :
+      by rw [lie_lmul_lmul_sq a, lie_lmul_lmul_sq b, lie_smul, lie_smul,
+        zero_add, add_zero]
+    ... = ⁅L a, L (b * b)⁆ + ⁅L b, L (a * a)⁆ + (2:ℤ)•⁅L a, L (a * b)⁆ + (2:ℤ)•⁅L b, L (a * b)⁆ :
+      by abel
+end
+
+/-- Linearise the operator form of the Jordan axiom (`lie_L_L_sq`) by substituting in a → a + b + c
+and expanding. When the Jordan axiom holds in all scalar extensions of `A`, an alternative proof is
+to substitute in a → a + λb + μc and equate coefficients of λμ.-/
+lemma two_lie_L_L_mul_add_lie_L_L_mul_add_lie_L_L_mul (a b c : A) :
```

Comment:
```suggestion
lemma two_nsmul_lie_lmul_lmul_add_add_eq_zero (a b c : A) :
```
(note that I have written `nsmul` rather than `zsmul` because I have also proposed below that we state this in terms of the natural number 2 instead of `2:ℤ`)